src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/StackOfTConverter.cs

```diff
@@ -14,6 +15,7 @@ protected override void Add(in TElement value, ref ReadStack state)
             ((TCollection)state.Current.ReturnValue!).Push(value);
         }
 
+        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor, typeof(Stack<>))]
```

Comment:
I'm not sure we should take this approach. In the PR to support all the collection types in source-gen mode, https://github.com/dotnet/runtime/pull/55566, these converters are being used by the source generator. That PR modifies them such that they perform no reflection/rooting (in source-gen mode), e.g. [this diff](https://github.com/dotnet/runtime/pull/55566/files#diff-6edf28878148d3d064b6b7cd591078671161a198ad430ba25c6f2425729f4298L38-L42).

I believe the plan is to replace the trimming tests that use the pre-existing serializer overloads with new ones that use the source generator instead - https://github.com/dotnet/runtime/issues/53437. Given this, could we disable the failing Stack test in this PR and replace it after https://github.com/dotnet/runtime/pull/55566 goes in? cc @eerhardt

Reply:
Agreed. I'd rather we just delete the trimming test until we have source gen support.

Do we know why this change is causing the test to fail though?